distributed/tests/test_client.py

```diff
@@ -5288,6 +5288,22 @@ def test_client_timeout_2():
         assert stop - start < 1
 
 
+@gen_test()
+def test_client_active_bad_port():
+    import subprocess
+    import shlex
+    if PY3:
+        p = subprocess.Popen(shlex.split('python -m http.server 8000'))
+    else:
+        p = subprocess.Popen(shlex.split('python -m SimpleHTTPServer 8000'))
```

Comment:
I'm curious, is it possible to do this from within python rather than shelling out?  Perhaps `from http.server import HTTPServer` and then something following?

Alternatively you could also take this as an opportunity to learn how to set up a Tornado web server, which is probably a good investment and would integrate more smoothly into the test suite.

http://www.tornadoweb.org/en/stable/httpserver.html#http-server

Reply:
I'm *sure* it's possible, but there at the least I would have to either not use `get_test` in order to handle the loop directly, or perhaps just try the same thing in a separate (normal) thread.

Reply:
You can get the loop directly by accessing the global singleton

```python
from tornado.ioloop import IOLoop
io_loop = IOLoop.current()
```

However this shouldn't be necessary.  All tornado code will get the current IOLoop by default.  You only need to manually specify an IOLoop if you are doing things in separate threads, which we shouldn't need to do here.